app/Http/Controllers/BookingRequestController.php

```diff
@@ -268,4 +270,54 @@ function ($attribute, $value, $fail) use ($request) {
     }
 
 
+    /**
+     * Filter booking requests by given json payload
+     *
+     * @param Request $request
+     * @return JsonResponse|Response
+     */
+    public function filter(Request $request)
+    {
+        // None of the request fields are mandatory, only
+        // filter the ones provided from request
+        $request->validate([
+            'status_list.*' => ['boolean'],
+            'date_range_start' => ['string'],
+            'date_range_end' => ['string'],
+            'data_reviewers' => ['array']
```

Comment:
double check if you'd want any of these to be a required field. I don't think so but worth a comment

Reply:
Don't want any of these as a required field. If the user accidentally click on filter without having changed a filter first, there should not be an error.

Reply:
yeah I figured that was the idea. sounds good, closing convo